test/boto_test.py

```diff
@@ -117,6 +115,59 @@ def test_credential_logging_empty(self):
                 key,
             )
 
+    def test_cli_arguments_check(self):
+        """
+        Boto warns users correctly when CLI arguments and passed in credentials don't match.
+        """
+        credential_map = {
+            ACCESS_KEY: {
+                'CLI': 'ZYXWVUTSR',
+                'ENV': 'ABCDEFGHI',
+                'msg': 'boto-access-key',
+            },
+            SECRET_KEY: {
+                'CLI': '0Z9Y8X7W6V5U4T3S2R1',
+                'ENV': '1A2B3C4D5E6F7G8H9I0',
+                'msg': 'boto-secret-key',
+            },
+        }
+
+        # Mocking the parser to trigger the warning
+        parser = krux.cli.get_parser(description=NAME)
+        add_boto_cli_arguments(parser)
+        namespace = parser.parse_args([
+            '--boto-access-key', credential_map[ACCESS_KEY]['CLI'],
+            '--boto-secret-key', credential_map[SECRET_KEY]['CLI'],
+        ])
+        mock_parser = MagicMock(
+            return_value=MagicMock(
+                spec=ArgumentParser,
+                autospec=True,
+                _action_groups=[],
+                parse_args=MagicMock(return_value=namespace)
+            )
+        )
+
+        # Mocking the logger to check for calls later
+        mock_logger = MagicMock(spec=Logger, autospec=True)
+
+        mock_env = {
+            ACCESS_KEY: credential_map[ACCESS_KEY]['ENV'],
+            SECRET_KEY: credential_map[SECRET_KEY]['ENV'],
+        }
+
+        with patch.dict('krux_boto.boto.os.environ', mock_env, clear=True):
+            with patch('krux_boto.boto.get_parser', mock_parser):
+                self.boto = Boto(
+                    logger=mock_logger
+                )
+
+        for key, value in credential_map.iteritems():
+            msg = 'You set %s as {0} in CLI, but passed %s to the library. Please check README to make sure you are passing the credentials correctly'.format(value['msg'])
+            cli_key = value['CLI'][0:3] + '[...]' + value['CLI'][-3:]
```

Comment:
I don't want this to give false positive when a horrible intern changes `_hide_value()` to return the string un-altered.